### PR TITLE
Jdrempel/serial connection guard

### LINKED ISSUE #15: Need a guard to prevent a single Test instance from locking up the serial line after it's done executing

### PR DESCRIPTION
**Pull request by jdrempel**
**Summary**

- Adds a guard to the `Test._run_full()` method that closes a Test's serial line after the test is executed, no matter what.

closes #15 

**Other Notes**

It may be beneficial to add better, more specific exception handling in the `Test._run_full()` method in the future but for now this works well enough.

### CODE REVIEW

**Comment by jdrempel** [comment]
closes #20

File: src/menus/config.py [hunk 1]
```diff
@@ -11,4 +11,4 @@
     1: "Back",
     9: "Exit",
     0: "Home",
-}
```
Comment (BradenStang) [inline src/menus/config.py:14]: Just wondering if this symbol means that previously before the change there was no new line?

Comment (jdrempel) [reply src/menus/config.py:14]: The red circle with the line through it means that there is no newline and the end of the file, which is considered a formatting error by the PEP8 standard. It's not a big deal here however.

**Review by BradenStang — APPROVED** [review]
I went through the code and it looks good to me, to the best of my knowledge. I did leave a comment closer to the beginning just asking about the no new line red symbol that appears a couple times throughout the code. I was also wondering, in some instances you had changed for example str=None, to str = None, and I was just wondering if the spacing makes a difference?

**Comment by jdrempel** [comment]
> I was also wondering, in some instances you had changed for example str=None, to str = None, and I was just wondering if the spacing makes a difference?

Technically no, Python will interpret it the same way, but the auto formatter that was added in this PR (`black`) enforces a certain spacing style for consistency and it bulldozed some of my earlier spacing choices. :robot:

**Pull request merged**